io/input/chunk_inputformat.cpp

```diff
@@ -34,8 +35,41 @@ ChunkInputFormat::ChunkInputFormat(int chunk_size) : chunk_size_(chunk_size) {
     is_setup_ = ChunkInputFormatSetUp::NotSetUp;
 }
 
+ChunkInputFormat::~ChunkInputFormat() {
+    ASSERT_MSG(splitter_ != nullptr, "Trying to destroy an illegal ChunkInputFormat object");
+    delete splitter_;
+    splitter_ = nullptr;
+}
+
 void ChunkInputFormat::set_input(const std::string& url) {
-    splitter_.load(url);
+    ASSERT_MSG(!is_setup(), "Last input not finished");

```

Comment:
You misunderstand this `SetUp` meaning. It's just for the configuration. You can not modify that in `next` function.